components/dashboards-web-component/src/designer/DashboardSettings.jsx

```diff
@@ -17,155 +17,224 @@
+import { FormPanel, Header } from '../common';
 import DashboardAPI from '../utils/apis/DashboardAPI';
+import DashboardSettingsRoles from './DashboardSettingsRoles';
+import { HttpStatus } from '../utils/Constants';
 
+/**
+ * Material UI theme.
+ */
 const muiTheme = getMuiTheme(darkBaseTheme);
+
+/**
+ * Style constants.
+ */
 const styles = {
-    messageBox: {textAlign: 'center', color: 'white'},
-    errorMessage: {backgroundColor: "#FF5722", color: "white"},
-    successMessage: {backgroundColor: "#4CAF50", color: "white"}
-};
-const httpStatus = {
-    ok: 200
+    messageBox: { textAlign: 'center', color: 'white' },
+    errorMessage: { backgroundColor: '#FF5722', color: 'white' },
+    successMessage: { backgroundColor: '#4CAF50', color: 'white' },
 };
 
+/**
+ * Dashboard settings.
+ */
 export default class DashboardSettings extends Component {
+    /**
+     * Constructor.
+     * Initilaizes the component state.
+     */
     constructor() {
         super();
         this.state = {
             dashboard: {},
             showMessage: false,
-            message: ''
+            message: '',
         };
         this.updateDashboard = this.updateDashboard.bind(this);
         this.showMessage = this.showMessage.bind(this);
     }
 
+    /**
+     * Loads the dashboar via the API.
+     */
     componentWillMount() {
         new DashboardAPI()
             .getDashboardByID(this.props.match.params.id)
-            .then(response => this.setState({dashboard: response.data}));
+            .then(response => this.setState({ dashboard: response.data }));
     }
 
+    /**
+     * Update dashboard.
+     */
+    updateDashboard() {
+        const { dashboard } = this.state;
+
+        // Validate fields.
+        if (dashboard.name.length === 0) {
+            this.showError('Please fill the dashboard name!');
+            return;
+        }
+
+        // Save dashboard metadata.
+        dashboard.pages = JSON.parse(dashboard.pages);
+        new DashboardAPI()
+            .updateDashboardByID(dashboard.url, dashboard)
+            .then((response) => {
+                if (response.status === HttpStatus.OK) {
+                    this.showMessage(`Dashboard '${dashboard.name}' is updated successfully.`);
+                    setTimeout(() => {
+                        window.location.href = window.contextPath;
+                    }, 1000);
+                }
+            })
+            .catch(() => this.showError('Failed updating dashboard roles.'));
+
+        // Save dashboard roles only if there are changes.
+        if (this.rolesComponent) {
+            this.rolesComponent
+                .saveRoles()
+                .then(() => this.showMessage('Dashboard roles saved successfully!'))
+                .catch(error => this.showError(error));
+        }
+    }
+
+    /**
+     * Show error message.
+     * 
+     * @param {string} message Error message.
+     */
+    showError(message) {
```

Comment:
can't we move this to a common util since its used in many places?

Reply:
Agreed! Let's handle that under separate issue.

Reply:
Created an issue https://github.com/wso2/carbon-dashboards/issues/659